sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1434,7 +1434,9 @@ class DataFrame private[sql](
     }
   }
 
-  private[sql] def internalRowRdd = queryExecution.executedPlan.execute()
+  // Note that the RDD is memoized. Once called, it won't change even if you change any
+  // query planning related Spark SQL configurations.
+  private[sql] def internalRowRdd = queryExecution.toRdd
```

Comment:
I guess I'm saying that I don't like this change.  Its just giving a new name to a concept that developers are already familiar with.  And now its possible that people will still use `toRdd` on the `queryExecution` (since its still there).  So, more concepts to juggle and changes to long lived concepts does not seem worth saving 6 characters to me.  Lazy vals from lazy vals also seems like unnecessary complexity to me.

IMHO, I think we should get rid of this new method instead.